sbin/interactive-config

```diff
@@ -0,0 +1,534 @@
+#!/bin/bash
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License version 2 as
+#  published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU General Public License for more details.
+
+BASEDIR=$(dirname $BASH_SOURCE)
+OUTPUT_DIR="saved_config"
+
+: ${FUNCTIONS_FILE="$BASEDIR/functions.sh"}
+## Load functions file
+if ! [ -e $FUNCTIONS_FILE ]
+then
+	echo "ERROR: Could not find function definitions (${FUNCTIONS_FILE})"
+	exit 1
+fi
+source $FUNCTIONS_FILE
+
+usage()
+{
+cat << EOF
+
+	interactive-config --output <output dir> --artifacts <artifacts dir> <target> <install_type>
+
+	interactive-config is a tool for user to interactively generate a usable config.
+
+	options:
+
+	--artifacts <artifacts_dir>: directory where the binary inputs for image
+	 	generation are found. The configuration script indicates what specific
+		images will be used.
+	--output <output_dir>: directory where the output configs will be used.
+		Defaults to \`pwd\`/saved_config.
+
+	<target>: The target device which will be installed.
+
+	- /dev/<x>: A physical USB device, or a network block device. This generates
+		a bootable USB installer that is capable of installing OverC onto
+		a device.
+
+	- <path>/<foo>.img: Generate a bootable image. In this mode the installer is
+		directly run to generate an image that is capable of being
+		written or flashed to a device and directly booted .. or
+		directly booted via qemu.
+
+	<install_type>: Check if this is an installer or a full image
+
+	- installer: will be a installer which will be installing on the final target
+	- full: a full image which will be running as the final system
+
+	For example:
+
+		# usb drive as sdc
+		$ interactive-config --artifacts `pwd` /dev/sdc
```

Comment:
Thanks for comments.

Currently the interactive config scripts only have ~500 lines of code, splitting it into a set of scripts not only increases the code/structure/maintenance complexity(probably only one dialog command per script), but also make it hard for flow control management. e.g. if we source and run the config files in the filename order, we might need to rename a certain set of scripts if we want to add something between two scripts in the middle of the series. Also each small scripts has different input/output which might need to handle differently, it is hard for the original source to provide when they just source it and run the default function.

So I suggest we can do that when we are advancing to a more complexed interactive installer, but to keep it simple as in one script for now.

Please let me know your thought.